test/buy-collateral-test.ts

```diff
@@ -562,5 +562,68 @@ describe('buyCollateral', function () {
       // EvilToken attack should be blocked, so totalSupplyBase should be 0
       expect(evilBobPortfolio.internal.EVIL).to.equal(0);
     });
+
+    it('reentrant supply is reverted', async () => {
```

Comment:
Should be buycollateral, not supply. same for comments in test

Reply:
Ok renamed to buyCollateral. :) 
Thought in the existing test is using 'supply', and decided to name it the same. 
(I guess that rationale with that was the retrant attack loop is happening in supplyFrom via buyCollateral)

Reply:
Oh I see. That means the test isn't testing the right function. It should be triggering reentrancy via `BuyCollateral`. I think you'll have to set it in the EvilToken.